src/payment/strategies/affirm/affirm-payment-strategy.spec.ts

```diff
@@ -0,0 +1,196 @@
+    let submitPaymentAction: Observable<Action>;
+    let store: CheckoutStore;
+    let strategy: AffirmPaymentStrategy;
+    let scriptLoader: AffirmScriptLoader;
+
+    beforeEach(() => {
+        const requestSender = createRequestSender();
+
+        affirm = getAffirmScriptMock();
+        affirm.checkout.open = jest.fn();
+        affirm.ui.error.on = jest.fn();
+        paymentMethod = getAffirm();
+        orderRequestSender = new OrderRequestSender(requestSender);
+        store = createCheckoutStore({
+            checkout: getCheckoutState(),
+            customer: getCustomerState(),
+            config: getConfigState(),
+            cart: getCartState(),
+            paymentMethods: getPaymentMethodsState(),
+            consignments: getConsignmentsState(),
+            billingAddress: getBillingAddressState(),
+        });
+        checkoutRequestSender = new CheckoutRequestSender(requestSender);
+        orderActionCreator = new OrderActionCreator(orderRequestSender, new CheckoutValidator(checkoutRequestSender));
+        paymentActionCreator = new PaymentActionCreator(
+            new PaymentRequestSender(createPaymentClient()),
+            orderActionCreator
+        );
+        paymentMethodActionCreator = new PaymentMethodActionCreator(new PaymentMethodRequestSender(requestSender));
+        scriptLoader = new AffirmScriptLoader();
+        strategy = new AffirmPaymentStrategy(
+            store,
+            orderActionCreator,
+            paymentActionCreator,
+            paymentMethodActionCreator,
+            scriptLoader
+        );
+
+        submitOrderAction = of(createAction(OrderActionType.SubmitOrderRequested));
+        submitPaymentAction = of(createAction(PaymentActionType.SubmitPaymentRequested));
+
+        jest.spyOn(orderActionCreator, 'submitOrder')
+            .mockReturnValue(submitOrderAction);
+
+        jest.spyOn(paymentActionCreator, 'submitPayment')
+            .mockReturnValue(submitPaymentAction);
+
+        jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod')
+            .mockReturnValue(paymentMethod);
+
+        jest.spyOn(paymentMethodActionCreator, 'loadPaymentMethod')
+            .mockResolvedValue(store.getState());
+
+        jest.spyOn(scriptLoader, 'load').mockResolvedValue(affirm);
+
+        payload = merge({}, getOrderRequestBody(), {
+            payment: {
+                methodId: paymentMethod.id,
+                gatewayId: paymentMethod.gateway,
+            },
+        });
+    });
+
+    describe('#initialize()', () => {
+        it('throws error if client token is missing', async () => {
+            jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod')
+                .mockReturnValue({
+                    ...paymentMethod,
+                    clientToken: null,
+                });
+
+            await expect(strategy.initialize({ methodId: paymentMethod.id })).rejects.toThrow(MissingDataError);
```

Comment:
So I think you either want to do this:

```ts
return expect(strategy.initialize({ methodId: paymentMethod.id }))
    .rejects.toThrow(MissingDataError);
```
(With the above, you don't need `async`)

or
```ts
try {
    await strategy.initialize({ methodId: paymentMethod.id })
} catch (error) {
    expect(error).toBeInstanceOf(MissingDataError);
}
```